packages/lib/src/helpers/encodeCall.ts

```diff
@@ -1,15 +1,35 @@
 import abi from 'ethereumjs-abi';
 import BN from 'bignumber.js';
 
-function formatValue(value: any): string {
-  if (typeof(value) === 'number' || BN.isBigNumber(value)) return value.toString();
-  else if (typeof(value) === 'string' && value.match(/\d+(\.\d+)?e(\+)?\d+/)) return (new BN(value)).toString(10);
-  else return value;
+export function formatValue(value: any): string | void {
+
+  // Numbers.
+  if(typeof(value) === 'number') {
+    if(value % 1 === 0) return value.toString(10);
+    else throw new Error('Floating point numbers are not supported on parameter encoding.');
+  }
+
+  // Big numbers.
+  if(BN.isBigNumber(value)) return value.toString();
+
+  // Strings.
+  if(typeof(value) === 'string') {
+    const hasHexRadix = value.indexOf('0x') !== -1 || value.indexOf('0X') !== -1;
+
+    // Numeric strings with exponents, e.g. '1.5e9'.
+    if(!hasHexRadix && value.match(/\d+(\.\d+)?e(\+)?\d+/)) {
```

Comment:
Aren't we doing the same mistake as before here, in trying to match just a part of the string, instead of all of it? Shouldn't replace the regex by something like `/^\s*\d+(\.\d+)?e(\+)?\d+\s*$/`? (note the start and end of string anchors, and the optional whitespace)